### ISSUE TITLE
Firestore triggers not firing in emulator mode for firebase functions v2

### ISSUE THREAD

**Issue by pavophilip** [firebase/firebase-functions#1381]
### Related issues
https://github.com/firebase/firebase-functions/issues/1213

### Version info

**node:** 16.5.0

**firebase-functions:** 4.3.0

**firebase-tools:** 11.28.0

**firebase-admin:** 11.7.0

### Test case

```
const functions = require('firebase-functions/v2');
const {onDocumentCreated} = require('firebase-functions/v2/firestore');

const admin = require('firebase-admin');
admin.initializeApp();

exports.oncreateproduct = onDocumentCreated("products", (event) => {
   console.log('oncreateproduct triggered');
});

exports.addproduct = functions.https.onCall(async (req, res) => {
    return {data: ''};
});

exports.addbrand = functions.https.onCall(async (req, res) => {

    return {data: ''};
});

```

### Steps to reproduce

1. Create functions.
2. Run emulators for Firestore and Functions.
3. Attempt to create a new product using the Emulator Suite or from the app using the SDK.

### Expected behavior

Expecting the function to be triggered when a new product is created.

### Actual behavior

- The function is not registered and does not appear in the logs when starting emulators. Only https callable initialized.
- The function does not trigger when creating a product.

Emulator logs:
```
✔  functions: Loaded functions definitions from source: addproduct, addbrand.
✔  functions[us-central1-addproduct]: http function initialized (http://127.0.0.1:5001/my-project/us-central1/addproduct).
✔  functions[us-central1-addbrand]: http function initialized (http://127.0.0.1:5001/my-project/us-central1/addbrand).

```

### Were you able to successfully deploy your functions?

Yes. It works properly in production mode.

**Comment by colerogers** [comment]
Hi @pavophilip you're correct that the emulator support for firestore hasn't shipped yet. There's still some changes that need to land before we can merge in https://github.com/firebase/firebase-tools/pull/5685. We hope to get it out shortly!

**Comment by ericeslinger** [comment]
It sure would have been great if there was some kind of documentation about this on the v2 functions doc. I eventually found this issue after about four or five hours trying to figure out if my typescript was wrong or my import/export stuff was wrong.

There's this one line at the bottom of [this doc](https://firebase.google.com/docs/functions/beta):
The Firebase Local Emulator Suite currently supports HTTP, Cloud Storage, and Pub/Sub functions in 2nd gen.

That's not particularly well-written. It would be a _lot_ better if the missing functionality in the emulator is directly called out on the new functions page (I decided to go with v2 since this is a side project so I don't mind the moving target api, but now I guess I'm going back to v1?)

Anyway: mitigation suggestion - update the docs [here](https://firebase.google.com/docs/emulator-suite/connect_functions#how_the_emulator_differs_from_production), which is where I looked first.
